pkg/sbom/nodelist.go

```diff
@@ -676,6 +676,21 @@ func (nl *NodeList) RelateNodeListAtID(nl2 *NodeList, nodeID string, edgeType Ed
 		nl.AddNode(n)
 	}
 
+	// Copy the remaining edges from n2
+	for _, e := range nl2.Edges {
+		// Check if we have an edge of the samer type already in the
+		// nodelist and if so, reuse it:
+		if _, ok := nlEdges[e.From]; ok {
+			if _, ok2 := nlEdges[e.From][e.Type]; ok2 {
+				nlEdges[e.From][e.Type][0].AddDestinationById(e.To...)
```

Comment:
Can you please explain why you are grabbing the first one out of this From.Type collection?

Reply:
In general, edges created by protobom should only have a single edge pairing from + type. Sometimes you'll have more if an SBOM has them defined.

This is taking the destination nodes from the SBOM2 edge and migrating them to the equivalent edge in SBOM1.

If there is more than one edge of the same type in SBOM1 (say from nodeX + CONTAINS), we don't care to which one we add the destinations from node2 as they are all the same logically, we even have a `cleanEdges()` function in the nodelist that collapses the duplicate edges when we need to.